lib/os_tty.c

```diff
@@ -5311,8 +5317,18 @@ ___device_tty *self;)
   ___SCMOBJ default_options =
     ___INT(___device_tty_default_options_virt (&d->base));
 
-  if (___device_kind (&d->base.base) == ___TTY_DEVICE_KIND ||
-      d->stage == TTY_STAGE_NOT_OPENED)
+  if (comint_under_emacs ())
```

Comment:
I don't like the name `comint_under_emacs`... perhaps `lineeditor_under_emacs_comint` ? My reasoning is that the function checks for the situation where Gambit's line editor is running under emacs' comint, and the behavioural consequences of that on the line editor.

Reply:
That'd be a fair name. I had initially changed that name's `lineeditor` part to `comint` because I was using that value to turn _off_ line editing mode, in which case the name `lineeditor_under_emacs` would have been misleading.

I kept the name with the latest commit because, conceptually, I think of "line editing" as a subset of "escape sequences" comprising actions like cursor movement, character deletion, etc.  The function's value is now intended to enable only the escape sequences not in my conceptual "line editing" set.  I also wanted to distinguish running Gambit from `term-mode` under Emacs as a different case.

From the perspective of the code as it exists right now, however, all features implemented via escape sequences, including stuff like colorization, are grouped under "line editor mode".  And so, from that perspective, `lineeditor_under_emacs_comint` would be the more appropriate name.  I think.